provisioning/baremetal_pod.go

```diff
@@ -280,14 +280,6 @@ func createContainerMetal3BaremetalOperator(images *Images, config *metal3iov1al
 			inspectorCredentialsMount,
 		},
 		Env: []corev1.EnvVar{
-			{
-				Name: "WATCH_NAMESPACE",
-				ValueFrom: &corev1.EnvVarSource{
```

Comment:
Question: what will be the impact, if any, on `LeaderElectionNamespace` for [BMO](https://github.com/metal3-io/baremetal-operator/blob/1432bfeab49722de48eaa35eef8279899c57c0fd/main.go#L114) in case of empty namespace?

Reply:
It appears that leader election needs a Namespace to be passed in https://github.com/kubernetes-sigs/controller-runtime/blob/0b554ebb54901a88427d4a89014af0d2dba1bbcf/pkg/manager/manager.go#L319.

With this change, BMO will not have the Namespace passed in via WATCH_NAMESPACE. But, with https://github.com/openshift/cluster-baremetal-operator/pull/107 we are passing in the POD_NAMESPACE which is the correct Namespace to be used for leader election. /cc @asalkeld